projects/scripts/adi_board.tcl

```diff
@@ -761,3 +761,81 @@ proc ad_cpu_interrupt {p_ps_index p_mb_index p_name} {
 ###################################################################################################
 ###################################################################################################
 
+# converting a string of ascii characters to a list of corresponding hex values
+proc char2hexlist {str} {
```

Comment:
Can we use something like this instead: https://wiki.tcl-lang.org/page/binary

Reply:
We are using "binary"

`# converting a string of ascii characters to a list of corresponding hex values
proc char2hexlist {str} {
  # string to int list

  >>>>> binary scan $str c* int_list

  # declare empty list
  set hex_list {}
  # convert int to hex and append to list
  foreach i $int_list {
    lappend hex_list [format %0.2X [expr {$i & 0xFF}]]
  }
  set hex_list;
}`

Reply:
I thought to implement the stringtohex conversion proc as in the second example on that wiki page :
`proc convert.string.to.hex str {`
`    binary scan $str H* hex`
`    return $hex `
`}`

Reply:
done